src/main/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -492,21 +494,30 @@ void enableFileSource(final SamReader reader, final boolean enabled) {
         return BAMFileSpan.merge(spanArray).toCoordinateArray();
     }
 
-    private class CRAMIntervalIterator extends BAMQueryMultipleIntervalsIteratorFilter
+    private abstract class CRAMIntervalIteratorBase extends BAMQueryMultipleIntervalsIteratorFilter
```

Comment:
it would be good to note here the reason for the abstractness, not only in line 513....or better, actually have an abstract method whose implementation will resolve the issue.

Reply:
Comment added. Not sure I see understand how the suggestion to add another abstract method resolves the initialize issue though. Can you give a more specific example ?